cli/config.go

```diff
@@ -4,65 +4,289 @@
+	CertsURL        string `toml:"certs_url"`
+	MsgContentType  string `toml:"msg_content_type"`
+	TLSVerification bool   `toml:"tls_verification"`
+}
+
+type filter struct {
+	Offset    string `toml:"offset"`
+	Limit     string `toml:"limit"`
+	RawOutput string `toml:"raw_output"`
 	Name      string `toml:"name"`
-	RawOutput bool   `toml:"raw_output"`
+	Contact   string `toml:"contact"`
+	Email     string `toml:"email"`
+	Metadata  string `toml:"metadata"`
+}
+
+type channel struct {
+	Status string `toml:"status"`
+	State  string `toml:"state"`
+	Topic  string `toml:"topic"`
+}
+
+type config struct {
+	Remotes remotes `toml:"remotes"`
+	Filter  filter  `toml:"filter"`
+	Channel channel `toml:"channel"`
 }
 
-// read - retrieve config from a file.
-func read(file string) (Config, error) {
-	data, err := os.ReadFile(file)
-	c := Config{}
+var (
+	errReadFail            = errors.New("failed to read config file")
+	errUnmarshalFail       = errors.New("failed to Unmarshall config TOML")
+	errConfigNotFound      = errors.New("config file was not found")
+	errUintConv            = errors.New("error converting filter to Uint64")
+	errBoolConv            = errors.New("error converting string to bool")
+	errUseExistConf        = errors.New("error using the existing configuration")
+	errNoKey               = errors.New("no such key")
+	errInvalidInt          = errors.New("error: invalid integer value for key")
+	errInvalidBool         = errors.New("error: invalid boolean value for key")
+	errUnsupportedKeyValue = errors.New("error: unsupported data type for key")
+	errMarshal             = errors.New("error marshaling the configuration")
+	errWritingConfig       = errors.New("error writing the updated config to file")
+	errInvalidURL          = errors.New("invalid url")
+	errURLParseFail        = errors.New("failed to parse url")
+	defaultConfigPath      = "./config.toml"
+)
+
+func init() {
+	if os.Getenv("GOBIN") != "" {
+		defaultConfigPath = os.Getenv("GOBIN") + "/config.toml"
+	}
+}
+
+func read(file string) (config, error) {
+	c := config{}
+	data, err := os.Open(file)
+	if err != nil {
+		return c, errors.Wrap(errReadFail, err)
+	}
+	defer data.Close()
+
+	buf, err := io.ReadAll(data)
 	if err != nil {
-		return c, errors.New(fmt.Sprintf("failed to read config file: %s", err))
+		return c, errors.Wrap(errReadFail, err)
 	}
 
-	if err := toml.Unmarshal(data, &c); err != nil {
-		return Config{}, errors.New(fmt.Sprintf("failed to unmarshal config TOML: %s", err))
+	if err := toml.Unmarshal(buf, &c); err != nil {
+		return config{}, errors.Wrap(errUnmarshalFail, err)
 	}
+
 	return c, nil
 }
 
-func ParseConfig() error {
+// Get config parameters from the config file.
+func ParseConfig(sdkConf mfxsdk.Config) (mfxsdk.Config, error) {
 	if ConfigPath == "" {
-		// No config file
-		return nil
+		ConfigPath = defaultConfigPath
+	}
+
+	_, err := os.Stat(ConfigPath)
+
+	// If the config file does not exist, create it.
+	if os.IsNotExist(err) {
```

Comment:
What if the error is not this one, i.e. the error is not nil, but also not NotExist?

Reply:
I see. Let me find out how to handle that

Reply:
Just an else, got it